File: AppSets-iOS/AppSets/com/app/appsets/im/MessageFromInfo.swift
//
//  MessageFromInfo.swift
//  AppSets
//
//  Created by caiju Xu on 2024/7/9.
//

import Foundation

struct MessageFromInfo : Bio {
    
    let uid:String
    var name: String? = nil
    var avatarUrl:String? = nil
    var roles:String? = nil
    
    var id: String{
        get{
            return uid
        }
    }
    
    var bioUrl: String? = nil
    
    init(uid: String, name: String?, avatarUrl: String?, roles: String? = nil) {
        self.uid = uid
        self.name = name
        self.avatarUrl = avatarUrl
        self.roles = roles
    }
}

File: AppSets-Android/appsets/src/main/kotlin/xcj/app/appsets/im/MessageFromInfo.kt
package xcj.app.appsets.im

import xcj.app.appsets.server.model.UserRole

data class MessageFromInfo(
    val uid: String,
    val name: String?,
    val avatarUrl: String? = null,
    val roles: String? = null
) : Bio {
    override val bioId: String
        get() = uid
    override val bioName: String?
        get() = name ?: uid
    override var bioUrl: Any? = null

    //ImMessage是否为系统发出的消息，非用户间的普通消息，规则是根据uid判断
    val isSystem: Boolean
        get() {
            return roles?.contains(UserRole.ROLE_ADMIN) == true
        }
}